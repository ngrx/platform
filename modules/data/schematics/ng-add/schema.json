{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxData",
  "title": "NgRx Data Schema",
  "type": "object",
  "properties": {
    "effect": {
      "type": "boolean",
      "default": true,
      "description":
        "When false use the EntityDataModuleWithoutEffects module instead of EntityDataModule."
    },
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description":
        "Do not add @ngrx/data as dependency to package.json (e.g., --skipPackageJson)."
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to the module.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "module": {
      "type": "string",
      "default": "app",
      "description": "Allows specification of the declaring module.",
      "alias": "m",
      "subtype": "filepath"
    },
    "migrateNgrxData": {
      "type": "boolean",
      "default": false,
      "description": "Migrate from ngrx-data, will rename modules.",
      "alias": "migrate"
    },
    "entityConfig": {
      "type": "boolean",
      "default": true,
      "description": "Create the Entity config file"
    }
  },
  "required": []
}
