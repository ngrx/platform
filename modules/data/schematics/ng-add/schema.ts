export interface Schema {
  path?: string;
  effects?: boolean;
  skipPackageJson?: boolean;
  project?: string;
  module?: string;
  migrateNgrxData?: boolean;
  entityConfig?: boolean;
}
