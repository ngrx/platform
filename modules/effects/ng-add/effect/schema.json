{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxEffect",
  "title": "NgRx Effect Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the effect.",
      "type": "string"
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "spec": {
      "type": "boolean",
      "default": true,
      "description": "Specifies if a spec file is generated."
    },
    "module": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring module.",
      "alias": "m",
      "subtype": "filepath"
    },
    "root": {
      "type": "boolean",
      "default": false,
      "description":
        "Flag to indicate whether the effects are registered as root."
    },
    "feature": {
      "type": "boolean",
      "default": false,
      "description": "Flag to indicate if part of a feature schematic."
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description": "Group effects file within 'effects' folder",
      "aliases": ["g"]
    }
  },
  "required": ["name"]
}
