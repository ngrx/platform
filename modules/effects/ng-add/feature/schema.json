{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxFeature",
  "title": "NgRx Feature State Options Schema",
  "type": "object",
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "name": {
      "description": "The name of the feature.",
      "type": "string"
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "module": {
      "type": "string",
      "description": "Specifies the declaring module.",
      "aliases": ["m"]
    },
    "spec": {
      "type": "boolean",
      "default": true,
      "description": "Specifies if a spec file is generated."
    },
    "reducers": {
      "type": "string",
      "description": "Specifies the reducers file.",
      "aliases": ["r"]
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description":
        "Group actions, reducers and effects within relative subfolders",
      "aliases": ["g"]
    }
  },
  "required": ["name"]
}
