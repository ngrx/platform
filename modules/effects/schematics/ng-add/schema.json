{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxRootEffect",
  "title": "NgRx Root Effect Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the effect.",
      "type": "string",
      "default": "App"
    },
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description":
        "Do not add @ngrx/effects as dependency to package.json (e.g., --skipPackageJson)."
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the effect.",
      "visible": false
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "spec": {
      "type": "boolean",
      "default": true,
      "description": "Specifies if a spec file is generated."
    },
    "module": {
      "type": "string",
      "default": "app",
      "description": "Allows specification of the declaring module.",
      "alias": "m",
      "subtype": "filepath"
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description": "Group effects file within 'effects' folder",
      "aliases": ["g"]
    }
  },
  "required": []
}
