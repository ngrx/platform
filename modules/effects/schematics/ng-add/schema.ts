export interface Schema {
  name: string;
  skipPackageJson?: boolean;
  path?: string;
  project?: string;
  flat?: boolean;
  spec?: boolean;
  module?: string;
  group?: boolean;
}
