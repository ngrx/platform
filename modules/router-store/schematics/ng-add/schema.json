{
  "$schema": "http://json-schema.org/schema",
  "$id": "SchematicsNgRxRouterStore",
  "title": "NgRx Router Store Schema",
  "type": "object",
  "properties": {
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description": "Do not add @ngrx/router-store as dependency to package.json (e.g., --skipPackageJson)."
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the router store.",
      "visible": false,
      "$default": {
        "$source": "workingDirectory"
      }
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "module": {
      "type": "string",
      "default": "app",
      "description": "Allows specification of the declaring module.",
      "alias": "m",
      "subtype": "filepath"
    }
  },
  "required": []
}
