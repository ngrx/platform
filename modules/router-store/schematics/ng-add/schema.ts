export interface Schema {
  skipPackageJson?: boolean;
  path?: string;
  project?: string;
  module?: string;
  standalone?: string;
}
