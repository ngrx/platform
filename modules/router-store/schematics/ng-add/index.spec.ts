import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import * as path from 'path';
import { Schema as RouterStoreOptions } from './schema';
import {
  getTestProjectPath,
  createWorkspace,
} from '@ngrx/schematics-core/testing';

describe('Router Store ng-add Schematic', () => {
  const schematicRunner = new SchematicTestRunner(
    '@ngrx/router-store',
    path.join(__dirname, '../collection.json')
  );
  const defaultOptions: RouterStoreOptions = {
    skipPackageJson: false,
    module: 'app',
  };

  const projectPath = getTestProjectPath();

  let appTree: UnitTestTree;

  beforeEach(async () => {
    appTree = await createWorkspace(schematicRunner, appTree);
  });

  it('should update package.json', async () => {
    const options = { ...defaultOptions };

    const tree = await schematicRunner
      .runSchematicAsync('ng-add', options, appTree)
      .toPromise();
    const packageJson = JSON.parse(tree.readContent('/package.json'));

    expect(packageJson.dependencies['@ngrx/router-store']).toBeDefined();
  });

  it('should skip package.json update', async () => {
    const options = { ...defaultOptions, skipPackageJson: true };

    const tree = await schematicRunner
      .runSchematicAsync('ng-add', options, appTree)
      .toPromise();
    const packageJson = JSON.parse(tree.readContent('/package.json'));

    expect(packageJson.dependencies['@ngrx/router-store']).toBeUndefined();
  });

  it('should be provided by default', async () => {
    const options = { ...defaultOptions };

    const tree = await schematicRunner
      .runSchematicAsync('ng-add', options, appTree)
      .toPromise();
    const content = tree.readContent(`${projectPath}/src/app/app.module.ts`);
    expect(content).toMatch(
      /import { StoreRouterConnectingModule } from '@ngrx\/router-store';/
    );
    expect(content).toMatch(/StoreRouterConnectingModule.forRoot\(\)/);
  });

  it('should import into a specified module', async () => {
    const options = { ...defaultOptions };

    const tree = await schematicRunner
      .runSchematicAsync('ng-add', options, appTree)
      .toPromise();
    const content = tree.readContent(`${projectPath}/src/app/app.module.ts`);
    expect(content).toMatch(
      /import { StoreRouterConnectingModule } from '@ngrx\/router-store';/
    );
  });

  it('should fail if specified module does not exist', async () => {
    const options = { ...defaultOptions, module: '/src/app/app.moduleXXX.ts' };
    let thrownError: Error | null = null;
    try {
      await schematicRunner.runSchematicAsync('ng-add', options, appTree);
    } catch (err) {
      thrownError = err;
    }
    expect(thrownError).toBeDefined();
  });
});
