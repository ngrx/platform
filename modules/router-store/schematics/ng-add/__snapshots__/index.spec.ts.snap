// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router Store ng-add Schematic Router Store ng-add Schematic for standalone application provides initial setup 1`] = `
"import { ApplicationConfig, provideZoneChangeDetection } from '@angular/core';
import { provideRouterStore } from '@ngrx/router-store';

export const appConfig: ApplicationConfig = {
  providers: [provideZoneChangeDetection({ eventCoalescing: true }), provideRouterStore()]
};
"
`;
